docstr_coverage/result_collection.py

```diff
@@ -0,0 +1,312 @@
+"""Module containing the classes required to collect and aggregate docstring information.
+
+Currently, this module is in BETA and its interface may change in future versions."""
+
+import enum
+import functools
+import operator
+from typing import Optional
+
+
+class ResultCollection:
+    """A result collection contains information about the presence of docstrings collected during
+    the walk through the files. From the `ResultCollection`, information about missing docstrings
+    for inspected files can be retrieved, and summary metrics (e.g. coverage) can be extracted
+    without having to walk over the files again."""
+
+    def __init__(self):
+        self._files = dict()
+
+    def get_file(self, file_path):
+        """Provides access to the docstring information for specific files.
+        This primarily targets information collection phase
+        (thus the docstr-coverage internal process).
+
+        If no file with the provided name is on record yet, a new empty `File` information instance
+        is created and returned.
+
+        Parameters
+        ----------
+        file_path: String
+            The path of the file for which the information is requested.
+
+        Returns
+        -------
+        File
+            The file (information) instance used to track docstring information"""
+        try:
+            return self._files[file_path]
+        except KeyError:
+            file = File()
+            self._files[file_path] = file
+            return file
+
+    def count(self):
```

Comment:
What do you think about renaming this to something like `make_aggregate`? `count` seems a little bit vague to me

Reply:
I don't like the `get` (it suggests that the method is a simple getter), so I changed the method to `count_aggregate`, which should hopefully resolve your rightful concerns.

Reply:
That's a good point! `count_aggregate` sounds great